pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -457,3 +471,80 @@ func (uc *UpgradeCluster) addVMToFinishedSets(vm compute.VirtualMachine, current
 		}
 	}
 }
+
+// checkControlPlaneNodesStatus checks whether it is safe to proceed with the upgrade process
+// by looking at the status of previously upgraded control plane nodes.
+//
+// It returns an error if more than 1 of the already-upgraded control plane nodes are in the NotReady state.
+// It returns nil if all control plane nodes are NotReady.
+// To recreate the node, users have to manually update the "orchestrator" tag on the VM.
+func (uc *UpgradeCluster) checkControlPlaneNodesStatus(ctx context.Context, upgradedNotReadyStream <-chan []string, mastersCount int) error {
+	if len(*uc.UpgradedMasterVMs) == 0 {
+		return nil
+	}
+	uc.Logger.Infoln("Checking status of upgraded control plane nodes")
+	upgradedNotReadyCount := 0
+loop:
+	for {
+		select {
+		case upgradedNotReady, ok := <-upgradedNotReadyStream:
+			if !ok {
+				break loop
+			}
+			upgradedNotReadyCount = len(upgradedNotReady)
+		case <-ctx.Done():
+			break loop
+		}
+	}
+	// return error if more than 1 upgraded node is not ready
+	// do not bother if upgradedNotReadyCount == mastersCount, at that point upgrade cannot take down more nodes
+	if upgradedNotReadyCount > 1 && upgradedNotReadyCount < mastersCount {
```

Comment:
Why the extra `&&` condition here? Shouldn't we always return error if there are more than one not ready control plane VM?

Reply:
you are right, I thought I had a good reason but on a second thought it is actually not